src/storage.rs

```diff
@@ -24,53 +27,282 @@ pub async fn init() -> sqlx::MySqlPool {
+            .await;
+        if let Err(err) = insert_res {
+            info!("Error during creating new account, {:#?}", err);
+            return Err(format!("{:#}", err))
+        };
+        let account: Vec<u8> = sqlx::query("SELECT (account) FROM accounts WHERE auth_pubkey = ?;")
+            .bind(auth_pubkey)
+            .fetch_one(self)
+            .await
+            .unwrap()
+            .get("account");
+        info!("Created account {:#?}, Adding auth_pubkey {:#?} as a default recipient_key to account", &account, &auth_pubkey);
+        sqlx::query("INSERT INTO recipients (account, recipient_key) VALUES (?, ?);")
+        .bind(&account)
+        .bind(auth_pubkey)
+        .execute(self)
+        .await
+        .unwrap();
+        Ok(())
+    }
+    // async fn vaporize_account(&self, auth_pubkey: String) {
+    //     let account: Vec<u8> = sqlx::query("SELECT (account) FROM accounts WHERE auth_pubkey = ?;")
+    //     .bind(&auth_pubkey)
+    //     .fetch_one(self)
+    //     .await
+    //     .unwrap()
+    //     .get("account");
+
+    //     let mut recipient_rows = sqlx::query(
+    //         "SELECT * FROM recipients WHERE account = ?;"
+    //     )
+    //         .bind(&account)
+    //         .fetch(self);
+
+    //     while let Some(recipient_row) = recipient_rows.try_next().await.unwrap() {
+    //         // map the row into a user-defined domain type
+    //         let recipient: Vec<u8> = recipient_row.get("recipient");  // binary decode
+    //         info!("Recipient {:x?}", recipient);
+    //         sqlx::query("DROP (*) FROM messages WHERE recipient = ?;")
+    //         .bind(&recipient)
+    //         .execute(self)
+    //         .await
+    //         .unwrap();
+    //         sqlx::query("DROP (*) FROM recipients WHERE recipient = ?;")
+    //         .bind(&recipient)
+    //         .execute(self)
+    //         .await
+    //         .unwrap();
+            
+    //     }
+    
+    // }
+    async fn persist_forward_message(&self, forward_msg: &ForwardMsg) -> Result<(), String> {
+        // Fetch recipients with given recipient_key
+        let mut rows = sqlx::query(
+            "SELECT * FROM recipients WHERE recipient_key = ?"
+        )
             .bind(&forward_msg.recipient_key)
+            .fetch(self);
+        // Save message for each recipient
+        while let Some(row) = rows.try_next().await.unwrap() {
+            // map the row into a user-defined domain type
+            let recipient: Vec<u8> = row.get("recipient");  // binary decode
+            info!("Persisting message for recipient {:x?}", recipient);
+            sqlx::query("INSERT INTO messages (recipient, message_data) VALUES (?, ?)")
+            .bind(&recipient)
             .bind(&forward_msg.message)
             .execute(self)
             .await
             .unwrap();
+        }
+        Ok(())
     }
     async fn retrieve_pending_message_count(&self, recipient_key: Option<&String>) -> u32 {
```

Comment:
this function too should have `auth_pubkey` argument - counting messages must be limited to the keys owned by a particular account

Reply:
if you follow my advice about `recipients` above, this function will as well greatly simplify

Reply:
> this function too should have `auth_pubkey` argument - counting messages must be limited to the keys owned by a particular account

Done.

Reply:
I have simplified the function, please have a look.  

https://github.com/nain-F49FF806/axum-test-server/pull/14/commits/9df1ef5bfe7caf9d96b26c0f0790ca2b0048776b

Turns out, it was unnecessarily complex regardless of the recipient situation (which I will tackle next). Using the `auth_pubkey` argumkent adds sufficient context to simply it greatly.